### PR TITLE
[issue-17: task 1-c]: Add model functions to connect DB functions and logical functions

### PR DESCRIPTION
**Pull request by a-ajith**
the PR is for task 1 of #17

PR contains DB connecting functions for ACI plugin

### CODE REVIEW

**Review by jeevan-kamkar — APPROVED** [review]

File: capmodel/common.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+import (
+	"fmt"
+	"github.com/ODIM-Project/ODIM/lib-utilities/common"
+	"github.com/ODIM-Project/ODIM/lib-utilities/errors"
+)
+
+const (
+	// TableFabric is the table for storing switch and pod ids
+	TableFabric = "pluginACIFabric"
```
Comment (Bharath-KKB) [inline capmodel/common.go]: get rid of plugin

File: capmodel/common.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	// TableSwitch is the table for storing switch information
+	TableSwitch = "pluginACISwitch"
+	// TableSwitchPorts is the table for storing ports of each switch
+	TableSwitchPorts = "pluginACISwitchPorts"
+	// TablePort is the table for storing port information
+	TablePort = "pluginACIPort"
+)
+
+// Fabric ACI holds data of switch id and pod id
+type Fabric struct {
```
Comment (Bharath-KKB) [inline capmodel/common.go:35]: There are a lot more than these two

File: capmodel/common.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		return fmt.Errorf("error while trying to connecting to DB: %v", err.Error())
+	}
+	if err = connPool.Create(table, key, data); err != nil {
+		return fmt.Errorf("error while trying to create new %v resource: %v", table, err.Error())
+	}
+	return nil
+}
+
+func readFromDB(table, key string) (string, *errors.Error) {
+	conn, err := common.GetDBConnection(common.OnDisk)
```
Comment (Bharath-KKB) [inline capmodel/common.go:56]: Should this be OnDIsk or InMemory

File: capmodel/getresource.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+package capmodel
+
+import (
+	"encoding/json"
+	dmtf "github.com/ODIM-Project/ODIM/lib-dmtf/model"
+	"github.com/ODIM-Project/ODIM/lib-utilities/errors"
+)
+
+// GetFabricFromDB will collect fabric data from Onisk DB
+func GetFabricFromDB(key string) (*Fabric, *errors.Error) {
```
Comment (Bharath-KKB) [inline capmodel/getresource.go:24]: What about collections

File: capmodel/common.go (hunk 2 above)
Comment (Bharath-KKB) [reply capmodel/common.go:35]: Also i see there are no unit tests.

**Review by Bharath-KKB — CHANGES_REQUESTED** [review]
Dont raise reviews unless you give a demo

File: db/operations.go [hunk 5]
```diff
@@ -54,21 +54,21 @@ func (c *Client) Create(table, key string, data interface{}) (err error) {
...
 	for {
-		var keys []string
-		var err error
-		keys, cursor, err = c.pool.Scan(cursor, generateKey(table, "*"), 100).Result()
+		keys, c, err := c.pool.Scan(cursor, generateKey(table, "*"), 100).Result()
 		if err != nil {
 			return nil, fmt.Errorf("failed to fetch all keys of table %s: %s", table, err.Error())
 		}
-		*allKeys = append(*allKeys, keys...)
+		allKeys = append(allKeys, keys...)
```
Comment (Bharath-KKB) [inline db/operations.go:65]: what is this change for?

File: db/operations.go [hunk 6]
```diff
@@ -54,21 +54,21 @@ func (c *Client) Create(table, key string, data interface{}) (err error) {
...
 // GetAllKeys will collect all the keys of provided table
-func (c *Client) GetAllKeys(table string) (allKeys *[]string, err error) {
+func (c *Client) GetAllKeys(table string) ([]string, error) {
+	var allKeys []string
 	var cursor uint64
 	for {
-		var keys []string
-		var err error
-		keys, cursor, err = c.pool.Scan(cursor, generateKey(table, "*"), 100).Result()
+		keys, c, err := c.pool.Scan(cursor, generateKey(table, "*"), 100).Result()
```
Comment (Bharath-KKB) [inline db/operations.go:61]: why are we scanning for all keys ? This is not acceptable

**Review by Bharath-KKB — CHANGES_REQUESTED** [review]
Address comments

**Pull request closed without merging**